metsim/methods/mtclim.py

```diff
@@ -106,127 +88,90 @@ def _calc_tfmax(precip, dtr, sm_dtr):
 
     # Calculate annual total precip
     sum_precip = df['precip'].values.sum()
-    ann_precip = (sum_precip / params['n_days']) * consts['DAYS_PER_YEAR']
+    ann_precip = (sum_precip / params['n_days']) * cnst.DAYS_PER_YEAR
     if ann_precip == 0.0:
         ann_precip = 1.0
 
     # Effective annual precip
     if params['n_days'] <= 90:
         # Simple scaled method, minimum of 8 cm 
         sum_precip = df['precip'].values.sum()
-        eff_ann_precip = (sum_precip / params['n_days']) * consts['DAYS_PER_YEAR']
+        eff_ann_precip = (sum_precip / params['n_days']) * cnst.DAYS_PER_YEAR
         eff_ann_precip = np.maximum(eff_ann_precip, 8.0)
         parray = eff_ann_precip
     else:
         # Calculate effective annual precip using 3 month moving window
-        parray = np.zeros(params['n_days'])
-        start_yday = df['day_of_year'][0]
-        end_yday = df['day_of_year'][-1]
-
+        window = np.zeros(params['n_days'] + 90)
+        window[90:] = df['precip']
+        
         # If yeardays at end match with those at beginning we can use
         # the end of the input to generate the beginning by looping around
         # If not, just duplicate the first 90 days
-        if start_yday != 1:
-            if end_yday == start_yday-1:
-                isloop = True
-        else:
-            if end_yday == 365 or end_yday == 366:
-                isloop = True
-        if isloop:
-            for i in range(90):
-                window[i] = df['precip'][params['n_days'] - 90 + i]
+        start_day, end_day = df['day_of_year'][0], df['day_of_year'][-1]
+        if (start_day%365 == (end_day%365)+1) or (start_day%366 == (end_day%366)+1):
+            window[:90] = df['precip'][-90:]
         else:
-            for i in range(90):
-                window[i] = df['precip'][i]
-        window[90:] = df['precip']
+            window[:90] = df['precip'][:90]
 
-        # FIXME: This can probably be vectorized
-        for i in range(params['n_days']):
-            sum_precip = 0.0
-            for j in range(90):
-                sum_precip += window[i+j]
-                sum_precip = (sum_precip / 90.) * consts['DAYS_PER_YEAR']
-            sum_precip = np.maximum(sum_precip, 8.0)
-            parray[i] = sum_precip
+        parray = np.array(pd.Series(window)
+                            .rolling(window=90, win_type=win_type,axis=0)
+                            .mean())[90:] * cnst.DAYS_PER_YEAR 
```

Comment:
2 pep comments: 1) space after `,`, 2) pep8 prefers the break after a `(` rather than at the period. In either event, the indentation is off here.

I think it also makes sense to keep parray as a Series.  Also, also, just fyi, anytime you need a numpy array out of a Series, just use the `.values` attribute.